docs/concepts/types.md

```diff
@@ -122,6 +122,49 @@ except ValidationError as exc:
     """
 ```
 
+#### Experimental pipeline API
+
+Pydantic v2.8.0 introduced an experimental pipeline API that allows composing parsing, constraints and transformations in a more type-safe manner than existing APIs. This API is subject to change or removal, we are looking for feedback and suggestions before making it a permanent part of Pydantic.
+
+```python
+from __future__ import annotations
+
+from datetime import datetime
+
+from typing_extensions import Annotated
+
+from pydantic import BaseModel
+from pydantic.transform_experimental import parse, parse_defer
+
+
+class User(BaseModel):
+    name: Annotated[str, parse(str).str.lower()]  # (1)!
+    age: Annotated[int, parse(int).gt(0)]  # (2)!
+    username: Annotated[str, parse(str).str.pattern(r'[a-z]+')]  # (3)!
+    password: Annotated[
+        str,
+        parse(str).transform(str.lower).predicate(lambda x: x != 'password')]  # (4)!
```

Comment:
Two observations here:

- It feels pretty verbose to have this pipeline inline. Should we encourage users to make it reusable?
  ```python
   validate_password = parse(str).transform(str.lower).predicate(lambda x: x != 'password')

   class User(BaseModel):
       password: Annotated[str, validate_password]
   ```

- Is there a way to not need the `Annotated`? I can't see an obvious one, but I'd love for it to work somehow. e.g. `Pipeline[str, transform(str.lower).predicate(...)]`

Reply:
`Pipeline[str, transform(str.lower).predicate(...)]` would work if we just export `Pipeline = Annotated` (I think) but that's not much better? I would love for `foo: parse(str).transform(int)` to work and infer the output type as int but that just doesn't work.

> Should we encourage users to make it reusable?

I agree, making examples reusable makes sense. I would suggest:

```python
# or call some password check/validation
Password = Annotated[str, parse(str).predicate(lambda x: x != 'password')]

 class User(BaseModel):
     password: Password
```